resources.Teams.ps1

```diff
@@ -0,0 +1,320 @@
+            $teamVisibility = "Private"
+
+            # Costcenter
+            $teamName = "$($resource.Name)"
+            # Sanitize team name, e.g. replace ' - ' with '_' or other sanitization actions 
+            $teamName = Get-ADSanitizeGroupName -Name $teamName
+
+            $teamDescription = "$($resource.Code)"
+            #endregion Change mapping here
+
+            $headers = New-AuthorizationHeaders -TenantId $AADtenantID -ClientId $AADAppId -ClientSecret $AADAppSecret
+            
+            $teamsFilter = "resourceProvisioningOptions/Any(x:x eq 'Team')"
+            $filter = "description+eq+'$($teamDescription)' & $teamsFilter"
+            Write-Verbose "Querying Azure AD group that matches filter '$($filter)'"
+
+            $baseUri = "https://graph.microsoft.com/"
+            $splatWebRequest = @{
+                Uri     = "$baseUri/v1.0/groups?`$filter=$($filter)&`$count=true"
+                Headers = $headers
+                Method  = 'GET'
+            }
+            $currentTeam = $null
+            $currentTeamResponse = Invoke-RestMethod @splatWebRequest -Verbose:$false
+            $currentTeam = $currentTeamResponse.Value
+
+            if ($null -eq $currentTeam.Id) {
+                $teamExists = $false
+            }
+            else {
+                $teamExists = $true
+            }
+
+            # If resource does not exist
+            if ($teamExists -eq $False) {
+                <# Resource creation preview uses a timeout of 30 seconds
+                while actual run has timeout of 10 minutes #>
+
+                $teamObject = [PSCustomObject]@{
+                    "template@odata.bind" = "https://graph.microsoft.com/v1.0/teamsTemplates('standard')"
+                    displayName           = $teamName
+                    description           = $teamDescription
+                    visibility            = $teamVisibility
+
+                    members               = [System.Collections.ArrayList]@()
+                }
+
+                if (-not[string]::IsNullOrEmpty($ownerAADUserId)) {
+                    [void]$teamObject.members.add(
+                        @{
+                            "@odata.type"     = "#microsoft.graph.aadUserConversationMember"
+                            roles             = @(
+                                "owner"
+                            )
+                            "user@odata.bind" = "https://graph.microsoft.com/v1.0/users('$ownerAADUserId')"
+                        }
+                    )
+                }
+
+                if (-Not($dryRun -eq $True)) {
+                    $headers = New-AuthorizationHeaders -TenantId $AADtenantID -ClientId $AADAppId -ClientSecret $AADAppSecret
+
+                    $body = $teamObject | ConvertTo-Json -Depth 10
+                    $baseUri = "https://graph.microsoft.com/"
+                    $splatWebRequest = @{
+                        Uri     = "$baseUri/v1.0/teams"
```

Comment:
for this specific endpoint permissions need to be set correctly. Please add this to the readme.

Reply:
Added to readme